From 102a161c76f8d4c8157390b7087ae6e65ec9d7cf Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Thu, 7 Jul 2022 06:18:12 +0300
Subject: [PATCH] build/gen.py: Make LAST_COMMIT_POSITION reproducible

Abbreviated hashes can vary in length depending on the number of objects
in the Git repository. This frequently breaks reproducibility for distro
packages. Address this by adding --abbrev=12 to the 'git describe' call.

Change-Id: I2fff11ff5a8e54e437a2186ea1f7c0ec17924437
---
 build/gen.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/gen.py b/build/gen.py
index fe56186b..f877cc24 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -218,8 +218,8 @@ def main(argv):
 def GenerateLastCommitPosition(host, header):
   ROOT_TAG = 'initial-commit'
   describe_output = subprocess.check_output(
-      ['git', 'describe', 'HEAD', '--match', ROOT_TAG], shell=host.is_windows(),
-      cwd=REPO_ROOT)
+      ['git', 'describe', 'HEAD', '--abbrev=12', '--match', ROOT_TAG],
+      shell=host.is_windows(), cwd=REPO_ROOT)
   mo = re.match(ROOT_TAG + '-(\d+)-g([0-9a-f]+)', describe_output.decode())
   if not mo:
     raise ValueError(
